### DISCUSSION TITLE
Validating order of fields in an object

### DISCUSSION THREAD

**Discussion by JensRantil** [cue-lang/cue#1896] [Q&A] [answered]
Hello!

I am using CUE to validate the JSON format of JSON objects from a third party vendor. In essence I'm using CUE to make sure I understand that format fully and I can apply my validation to hundreds of these JSON files. This works really really great!

The problem is that in production these JSON objects can be very large (gigabytes). As such, I will need to do some streaming parsing in production to be comfortable I'm not hitting memory limits. Streaming parsing requires I understand the order of the JSON objects' fields. An object essentially looks like this:
```
{
   "smallProperty1": 42,
   "smallProperty2": 84,
   "largeProperty": [
      // ginormeous array
   ]
}
```
Is there a way for me to validate that `largeProperty` is the last field in a JSON object? Can I be certain that a list comprehension over an object will return the keys in the same order as they are given in input? Other ways?

**Comment by mvdan** [comment, marked as answer]
Hello @JensRantil! I realise this reply comes a tad late, but I believe you're after https://github.com/cue-lang/cue/issues/474. Right now, iterating over the fields in a struct via a comprehension has an unspecified order. If the order were specified, then you could use that iteration to get the field indexes, and enforce that the index of `largeProperty` is equal to the length of the parent object minus one.

**Reply by JensRantil** [reply]
Thanks @mvdan! That issue very well reflects the issue I'm facing. I'll keep an eye on that issue.
